sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/SensorContextUtilsTest.java

```diff
@@ -157,11 +161,108 @@ public void hasFilesOfLanguage_whenOnlyOtherLanguageExists_returnsFalse() {
+    var inputFile = new TestInputFileBuilder("mod", "source.cs")
+      .setLanguage("cs")
+      .setType(Type.MAIN)
+      .setContents("\tSome text\nrangeStartingAtEOL1\nrangeStarting\nAtEOL2\nAndSpanning\nMultipleLines")
+      .build();
+    fs.add(inputFile);
+
+    // Possible real scenario: 4 spaces transformed into \t -> new EOL at 10 instead of 13
+    assertTextRange(toTextRange(inputFile, pbTextRangeOf(1, 13, 2, 18)), 2, 0, 2, 18);
+  }
+
+  @Test
+  public void toTextRange_whenMultiLineRangeEndsBeyondEOL_trimsBasedOnEndLineLength() {
+    var inputFile = new TestInputFileBuilder("mod", "source.cs")
+      .setLanguage("cs")
+      .setType(Type.MAIN)
+      .setContents("Some text multiline\nRangeWithEndLineOffsetBiggerThanStartLineOffset Some other text")
+      .build();
+    fs.add(inputFile);
+    assertTextRange(toTextRange(inputFile, pbTextRangeOf(1, 10, 2, 46)), 1, 10, 2, 46);
+  }
+
+  @Test
+  public void toTextRange_whenSingleLineRangeStartsAtEOL_filtersOut() {
+    var inputFile = new TestInputFileBuilder("mod", "source.cs")
+      .setLanguage("cs")
+      .setType(Type.MAIN)
+      .setContents("Some text\nSome other text")
+      .build();
+    fs.add(inputFile);
+    assertThat(toTextRange(inputFile, pbTextRangeOf(1, 9, 1, 12))).isEmpty();
+    assertThat(toTextRange(inputFile, pbTextRangeOf(1, 9, 1, 9))).isEmpty();
+  }
+
+  @Test
+  public void toTextRange_whenSingleLineRangeStartsBeyondEOL_filtersOut() {
+    var inputFile = new TestInputFileBuilder("mod", "source.cs")
+      .setLanguage("cs")
+      .setType(Type.MAIN)
+      .setContents("Some text\nSome other text")
+      .build();
+    fs.add(inputFile);
+    assertThat(toTextRange(inputFile, pbTextRangeOf(1, 10, 1, 12))).isEmpty();
+    assertThat(toTextRange(inputFile, pbTextRangeOf(1, 10, 1, 10))).isEmpty();
+  }
+
+  @Test
+  public void toTextRange_whenSingleLineRangeEndsBeyondEOL_trimsBasedOnEndLineLength() {
+    var inputFile = new TestInputFileBuilder("mod", "source.cs")
+      .setLanguage("cs")
+      .setType(Type.MAIN)
+      .setContents("Some text singleLineRange\n")
+      .build();
+    fs.add(inputFile);
+    assertTextRange(toTextRange(inputFile, pbTextRangeOf(1, 10, 1, 100)), 1, 10, 1, 25);
+  }
+
   private void addFileToFileSystem(String fileName, InputFile.Type fileType, String language) {
     DefaultInputFile inputFile = new TestInputFileBuilder("mod", fileName)
       .setLanguage(language)
       .setType(fileType)
       .build();
     fs.add(inputFile);
   }
+
+  private SonarAnalyzer.TextRange pbTextRangeOf(int startLine, int startLineOffset, int endLine, int endLineOffset) {
+    return SonarAnalyzer.TextRange.newBuilder()
+      .setStartLine(startLine)
+      .setStartOffset(startLineOffset)
+      .setEndLine(endLine)
+      .setEndOffset(endLineOffset)
+      .build();
+  }
+
+  private void assertTextRange(Optional<TextRange> textRange, int startLine, int startLineOffset, int endLine, int endLineOffset) {
```

Comment:
@costin-zaharia-sonarsource  While the de-facto standard (in Java and JUnit5 too) is to have the expected first and the actual after, I have inverted here because there are 4 different expected values, and it seems a bit off to have the actual as 5th parameter.
I can change it, if we want to stick to the convention. WDYT?

Reply:
I think it's fine, I would not stress about it.